io/io/src/TFile.cxx

```diff
@@ -306,6 +306,16 @@ TFile::TFile() : TDirectoryFile(), fCompress(ROOT::RCompressionSetting::EAlgorit
 TFile::TFile(const char *fname1, Option_t *option, const char *ftitle, Int_t compress)
            : TDirectoryFile(), fCompress(compress), fUrl(fname1,kTRUE)
 {
+   auto handleZombie = [this] {
+      // error in file opening occurred, make this object a zombie
+      {
+         R__LOCKGUARD(gROOTMutex);
+         gROOT->GetListOfClosedObjects()->Add(this);
```

Comment:
The lock is redundant here.  the list of ClosedObject is a thread safe collection.

Reply:
This comment is a duplicate of: https://github.com/root-project/root/pull/11719